sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -887,14 +906,15 @@ private async Task RunAsync(CancellationToken cancellationToken)
                     completeOwnershipList = (await StorageManager.ListOwnershipAsync(FullyQualifiedNamespace, EventHubName, ConsumerGroup, cancellationToken)
                         .ConfigureAwait(false))
                         .ToList();
+                    Logger.CompleteOwnershipRecordsCount(completeOwnershipList.Count());
```

Comment:
since completeOwnershipList is a List, we can use `completeOwnershipList.Count`

Reply:
I think I don't get this. Could you please tell me what is the concern here?

I'm already using `completeOwnershipList.Count()`.

Reply:
It's really a nit, but `Count()` is an extension method on `IEnumerable<T>`, versus `Count`, which is a property on the `ICollection<T>` interface, which `List<T>` implements.

Effectively [they are equivalent in this case](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Linq/src/System/Linq/Count.cs#L21), as `Count()` optimizes to evaluate the `Count` property if the `IEnumerable<T>` also implements `ICollection<T>`. 

The reason it tends to catch people's attention is that if `completeOwnership` were just a basic `IEnumerable` then `Count()` would cause an enumeration of the collection rather than immediately returning the internal count.

Reply:
I understand the difference here, but `completeOwnershipList` is a `IEnumerable<T>` and therefore it doesn't has a `count` property, and hence we'd have to use `count()` method unless we typecast it to `ICollection<T>` which I think would be overkill for this small usecase. And as you said, `count()` method is optimized for `ICollection<T>` usecase, I would like to keep the current impl as is unless you've strong concerns here.

Reply:
Sorry, you are correct- I saw the.`ToList()` and totally missed the definition right above :)

Reply:
I'd typically say that there is benefit to creating an intermediate to capture the list so that we're not calling `Count()` - but since these lines have good locality, it is reasonable to say that those working in here would be able to see that `ToList()` has been called so the enumerable should use the optimization for `Count()` and is safe to walk multiple times.

If these start to drift apart, we may want to consider at least a comment calling out that we know it's a list.